### DISCUSSION TITLE
Reference to icon font name instead of unicode

### DISCUSSION THREAD

**Discussion by bohorok** [polybar/polybar#2966] [General]
Each icon in the glyph table has a name. For example, the lowercase letter a is: "Latin small letter a". I created my table of glyphs, and named them accordingly. Can I refer to these glyphs by their name in the config file.
thanks in advance for your answer.

**Comment by patrick96** [comment]
I am not sure what exactly you want to do here. But in general, getting values into the config file from the outside is done using [references](https://github.com/polybar/polybar/wiki/Configuration#referencing-values). Environment variables and xrdb values will be most useful here. Unfortunately these references are quite limited: they cannot be mixed with other text, so something like `label = ${env:ICON_VOLUME} %percentage%%` does **not** work.

What I do in my config file is, for all values where I have this problem, I do this string manipulation in my bash script that starts polybar. So for the example above, I have `label = ${env:LABEL_VOLUME}` and in the bash script I do `export LABEL_VOLUME="$ICON_VOLUME %percentage%%"`.

**Comment by bohorok** [comment]
Sorry for the late reply. 

> I am not sure what exactly you want to do  here.

I created my own font-Icons popular applications and
I did something like the dock:

`##################### MODULE DOCK APPS ########################

[module/dock-apps]
type = custom/text
content-padding = 3pt
content-background = ${colors.color-9a}
content-font = 2
content = :: %{A1:brave:}a%{A} %{A1:tilix:}c%{A} %{A1:subl:}b%{A} %{A1:doublecmd:}d%{A} %{A1:gpick:}e%{A} %{A1:fsearch:}g%{A} %{A1:speedcrunch:}p%{A} ::`

In line content, instead of writing a, b, d etc. , to refer  to icon, it is better to use the names of the icons that I gave when creating them, i.e. 'brave', 'tilix', etc.
